hack/procs/multiWorker.ml

```diff
@@ -38,57 +38,52 @@ let single_threaded_call_dynamic job merge neutral next =
   done;
   !acc
 
+let multi_threaded_call_dynamic
```

Comment:
> hmm so why was this refactoring necessary? Just so we keep around the handles?

Yes. But I changed it more than required. I removed a very small code
duplication (the exception handler for `Exit` is not required
anymore). I might simplify the "diff" and stand closer to the actual
code, if you prefer.